src/js/components/Tag/Tag.js

```diff
@@ -0,0 +1,75 @@
+import React, { forwardRef, useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+
+import { FormClose } from 'grommet-icons/icons/FormClose';
+
+import { defaultProps } from '../../default-props';
+
+import { TagPropTypes } from './propTypes';
+import { Box } from '../Box';
+import { Text } from '../Text';
+
+import { StyledRemoveButton, StyledTagButton } from './StyledTag';
+
+const Tag = forwardRef(
+  ({ name, value, onRemove, onClick, ...rest }, ref) => {
+
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+    
+    const containerProps = {
+      ref,
+      align: 'center',
+      background: theme.tag.background,
+      border: theme.tag.border,
+      round: theme.tag.round,
+      ...rest,
+    };
+    const contents = (
+      <Box width={{ min: 'min-content' }} pad={theme.tag.pad}>
+        <Text>
+          {name && <Text {...theme.tag.name}>{name}</Text>}
+          {name && value ? theme.tag.separator : '' }
+          {value && <Text {...theme.tag.value}>{value}</Text>}
+        </Text>
+      </Box>
+    );
+
+    return onRemove || !onClick ? (
```

Comment:
If I'm reading this correctly, this allows Tag to have `onRemove` and `onClick` which would create a nested interactive element. Do we want to allow this?

Reply:
We don't allow both. onClick will be ignored if onRemove is specified.

onClick is only used at line 61. If they specify onRemove it will do the static label plus a remove button. Same thing if neither onClick or onRemove are specified.

If only onClick is specified it will render a StyledTagButton with the onClick callback included.

I've written the component doc for grommet-site to say you can't use both. This may deserve a console.warn similar to what we do in a couple other components though